rskj-core/src/test/java/co/rsk/peg/RegisterBtcTransactionIT.java

```diff
@@ -32,58 +34,108 @@
+    private BtcBlockStoreWithCache.Factory btcBlockStoreFactory;
+    private BtcBlockStoreWithCache btcBlockStoreWithCache;
+    private BtcTransaction bitcoinTransaction;
+    private PartialMerkleTree pmtWithTransactions;
+    private int btcBlockWithPmtHeight;
+    private int chainHeight;
+    private Transaction rskTx;
+    private ECKey ecKey;
+    private RskAddress rskReceiver;
+    private BridgeSupport bridgeSupport;
+
+    @BeforeEach
+    void setUp() {
+        rskTx = TransactionUtils.createTransaction();
+        repository = BridgeSupportTestUtil.createRepository();
+        track = repository.startTracking();
+
+        activationConfig = ActivationConfigsForTest.all().forBlock(0);
+        rskExecutionBlock = getRskExecutionBlock();
+
+        btcLockSenderProvider = new BtcLockSenderProvider();
+        feePerKbSupport = getFeePerKbSupport();
 
         Federation federation = P2shErpFederationBuilder.builder().build();
         FederationStorageProvider federationStorageProvider = getFederationStorageProvider(track, federation);
-        FederationSupport federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
+        federationSupport = getFederationSupport(federationStorageProvider, activationConfig, bridgeConstants.getFederationConstants());
+
+        bridgeStorageProvider = new BridgeStorageProvider(track, PrecompiledContracts.BRIDGE_ADDR, bridgeConstants.getBtcParams(), activationConfig);
+        btcBlockStoreFactory = new RepositoryBtcBlockStoreWithCache.Factory(bridgeConstants.getBtcParams(), 100, 100);
+        btcBlockStoreWithCache = btcBlockStoreFactory.newInstance(track, bridgeConstants, bridgeStorageProvider, activationConfig);
 
         BtcECKey btcPublicKey = new BtcECKey();
-        Coin btcTransferred = Coin.COIN;
-        BtcTransaction bitcoinTransaction = createPegInTransaction(federationSupport.getActiveFederation().getAddress(), btcTransferred, btcPublicKey);
+        ecKey = ECKey.fromPublicOnly(btcPublicKey.getPubKey());
+        rskReceiver = new RskAddress(ecKey.getAddress());
+        bitcoinTransaction = createPegInTransaction(federationSupport.getActiveFederation().getAddress(), BTC_TRANSFERRED, btcPublicKey);
+
+        pmtWithTransactions = createValidPmtForTransactions(Collections.singletonList(bitcoinTransaction.getHash()), bridgeConstants.getBtcParams());
+        btcBlockWithPmtHeight = bridgeConstants.getBtcHeightWhenPegoutTxIndexActivates() + bridgeConstants.getPegoutTxIndexGracePeriodInBtcBlocks();
+        chainHeight = btcBlockWithPmtHeight + bridgeConstants.getBtc2RskMinimumAcceptableConfirmations();
+
+        recreateChainFromPmtOrFail();
+        bridgeStorageProvider.save();
+        bridgeSupport = getBridgeSupport(bridgeStorageProvider, activationConfig, federationSupport, feePerKbSupport, rskExecutionBlock, btcBlockStoreFactory, track, btcLockSenderProvider);
```

Comment:
I would recommend to just use the builder here as well. This way we reduce this big param list.